spec/registry.md

```diff
@@ -347,6 +347,14 @@ A [Number Operand](#number-operands) without a `currency` _option_ results in a
 > For example, in ICU4J, the type `com.ibm.icu.util.CurrencyAmount` can be used
 > to set the amount and currency.
 
+> [!NOTE]
+> The `currency` is only required to be well-formed rather than checked for validity.
+> This allows new currency codes to be defined 
+> (there are many recent examples of this occuring).
+> It also avoids requiring implementations to check currency codes for validity,
+> although implementations are permitted to emit _Bad Option_ or _Bad Operand_ for invalid codes.
+> A well-formed currency code matches the regular expression `^[A-Z]{3}$`.
```

Comment:
The source on 341 says:

> Codes consisting of 3 ASCII letters that are or have been valid in ISO 4217, plus certain additional codes that are or have been in common use. The list of countries and time periods associated with each currency value is available in [Supplemental Currency Data](https://cldr-smoke.unicode.org/spec/main/ldml/tr35-numbers.html#Supplemental_Currency_Data), plus the default number of decimals.
>
> The XXX code is given a broader interpretation as Unknown or Invalid Currency.

This doesn't distinguish valid and well-formed. I suppose we might link [ISO4217](https://www.iso.org/standard/64758.html)'s "about" page. Not sure anyone defines well-formed as a distinct entity.

In practice, this defines an alpha3 code. I should probably use ABNF (since we use that everywhere else in our spec) to define it. I'll provide an edit.